### PR TITLE
add tic tac toe

### PR DESCRIPTION
**Pull request by brianjohnson5972**
#2214 
Added tic tac toe into the build, brought it up to date with contract API and added unit test to ensure it stays up to date with contract API.

### CODE REVIEW

File: tests/wasm_tests/tic_tac_toe_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+};
+
+BOOST_AUTO_TEST_SUITE(tic_tac_toe_tests)
+
+BOOST_AUTO_TEST_CASE( tic_tac_toe_game ) try {
+   TESTER chain;
+   abi_serializer abi_ser = json::from_string(tic_tac_toe_abi).as<abi_def>();
+
+   chain.set_code(config::system_account_name, eosio_system_wast);
+   chain.set_abi(config::system_account_name, eosio_system_abi);
```
Comment (heifner) [inline tests/wasm_tests/tic_tac_toe_tests.cpp]: Do you need to set the system account for this to work?

File: tests/wasm_tests/tic_tac_toe_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+         ("challenger", "player2")
+         ("host", "player1")
+         ("by", "player2")
+         ("mvt", mvt)
+      ), transaction_exception, assert_message_contains("the game has ended"));
+
+   chain.push_action(N(tic.tac.toe), N(close), N(player1), mutable_variant_object()
+           ("challenger", "player2")
+           ("host", "player1")
+   );
```
Comment (heifner) [inline tests/wasm_tests/tic_tac_toe_tests.cpp]: Can you add step to verify winner.

File: tests/wasm_tests/tic_tac_toe_tests.cpp (hunk 1 above)
Comment (brianjohnson5972) [reply tests/wasm_tests/tic_tac_toe_tests.cpp]: No, removed

File: tests/wasm_tests/tic_tac_toe_tests.cpp (hunk 2 above)
Comment (brianjohnson5972) [reply tests/wasm_tests/tic_tac_toe_tests.cpp]: Yes, also added Tester methods to access contract table data.

**Pull request merged**